atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -2551,6 +2560,83 @@ public void exceptionThrownWhenTooManyPostFilterIterationsOccur() {
                         SafeArg.of("maxIterations", SnapshotTransaction.MAX_POST_FILTERING_ITERATIONS));
     }
 
```

Comment:
I also moved Tests like `metadataStoredAfterPutWithMetadata` to `LocalWriteBufferTest` as they are essentially testing the buffer behavior. This is also neat in the sense that we no longer need to expose the local write buffer for metadata tests.